core/ibft.go

```diff
@@ -797,7 +799,7 @@ func (i *IBFT) handlePrepare(view *proto.View, quorum uint64) bool {
 		isValidPrepare,
 	)
 
-	if len(prepareMessages) < int(quorum)-1 {
+	if i.backend.HasQuorum(view, prepareMessages) {
```

Comment:
Negate this condition, after you change `HasQuorum` implementation in the `MockBackend`:

```suggestion
	if !i.backend.HasQuorum(view, prepareMessages) {
```